include/dxc/Support/dxcapi.extval.h

```diff
@@ -0,0 +1,31 @@
+#include "dxc/Support/dxcapi.use.h"
+#include <cassert>
+#include <string>
+
+namespace dxc {
+class DxcDllExtValidationLoader : public DllLoader {
+  // DxCompilerSupport manages the
+  // lifetime of dxcompiler.dll, while DxilExtValSupport
+  // manages the lifetime of dxil.dll
+  dxc::SpecificDllLoader DxCompilerSupport;
+  dxc::SpecificDllLoader DxilExtValSupport;
+
+  std::string DxilDllPath;
+  HRESULT InitializeInternal(LPCSTR fnName);
+
+public:
+  std::string GetDxilDllPath() { return DxilDllPath; }
+  bool DxilDllFailedToLoad() {
+    return !DxilDllPath.empty() && !DxilExtValSupport.IsEnabled();
+  }
+
+  HRESULT CreateInstanceImpl(REFCLSID clsid, REFIID riid,
+                             IUnknown **pResult) override;
+  HRESULT CreateInstance2Impl(IMalloc *pMalloc, REFCLSID clsid, REFIID riid,
+                              IUnknown **pResult) override;
+
+  HRESULT Initialize() { return InitializeInternal("DxcCreateInstance"); }
```

Comment:
Why does `InitializeInternal` take a `fnName` argument if it's only ever called with a fixed string? Do we need `InitializeInternal` at all, or should we just update the implementation to be an implementation of `Initialize()`?

Reply:
Good point, looks like we dont need it at all. I'll remove InitializeInternal and move its implementation to Initialize().